src/GafferImageUI/ImageView.cpp

```diff
@@ -393,12 +432,86 @@ bool ImageView::keyPress( const GafferUI::KeyEvent &event )
 				);
 				return true;
 			}
+			if(event.key == "Home")
+			{
+				
+				V2i viewport = viewportGadget()->getViewport();
+				V2i viewportCenter(viewport.x / 2, viewport.y / 2);
+				Box3f bound = m_imageGadget->bound();
+				V2i imageCenter((bound.max.x - bound.min.x) / 2, (bound.max.y - bound.min.y) / 2);
+				Box3f frame(
+					Imath::V3f(imageCenter.x - viewportCenter.x, imageCenter.y - viewportCenter.y, 0),
+					Imath::V3f(imageCenter.x + viewportCenter.x, imageCenter.y + viewportCenter.y, 0)
+				);
+				viewportGadget()->frame(frame);
+				return true;
+			}
+            if(event.key == "Plus")
+            {
+                float zoomLevel = computeZoomLevel();
+                float roundLevel;
+
+                if((zoomLevel + EPSILON) < 1.f)
+                {
+                    zoomLevel = std::min(100.f, 1 / zoomLevel);
+                    roundLevel = round_down_to_even(zoomLevel);
+                    
+                    if(cmpf(zoomLevel, roundLevel, zoomLevel * .1))
+                    {
+                        roundLevel = roundLevel * .5;
+                    }
+                    roundLevel = 1 / roundLevel;
+                }
+                else
+                {
+                    roundLevel = round_up_to_even(zoomLevel);
+                    if(cmpf(zoomLevel, roundLevel, zoomLevel * .1))
+                    {
+                        roundLevel = roundLevel * 2;
+                    }
+                }
+                zoomLevelPlug()->setValue(roundLevel) ;
+				
+                return true;
+
+            }
+            if(event.key == "Minus")
+            {
+                float zoomLevel = computeZoomLevel();
+                float roundLevel;
+                if((zoomLevel - EPSILON) < 1.f)
+                {
+                    zoomLevel = std::min(1000.0, round(1 / zoomLevel));
+                    roundLevel = round_up_to_even(zoomLevel);
+                    if(cmpf(zoomLevel, roundLevel, zoomLevel * .1))
+                    {
+                        roundLevel = roundLevel * 2;
+                    }
+                    roundLevel = 1 / roundLevel;
+                }
+                else
+                {
+                    roundLevel = round_down_to_even(round(zoomLevel));
+                    if(cmpf(zoomLevel, roundLevel, zoomLevel * .1))
+                    {
+                        roundLevel = roundLevel * .5;
```

Comment:
Perhaps you could start by clearly describing the behaviour you want to achieve, and then we can come up with the code to achieve it? For instance, I just experimented with +/- in Inkscape, and the behaviour I saw was this :

- Zoom is specified as a percentage
- Hitting + or - adjusts the zoom by a factor of roughly 1.41 each time
- There is no "snapping", so if you zoom in a single step from 70% for instance, you end up on 99% - there is no attempt to snap up to 100% automatically.

I'm not saying we should follow that exactly, just that I think we need to define our desired behaviour in similar terms first, before we discuss code.